fastapi/models/fastapi_endpoint.py

```diff
@@ -0,0 +1,268 @@
+            return None
+        return ASGIMiddleware(record._get_app())
+
+    @api.model
+    @tools.ormcache("root_path")
+    def get_uid(self, root_path):
+        record = self.search([("root_path", "=", root_path)])
+        if not record:
+            return None
+        return record.user_id.id
+
+    def _get_app(self) -> FastAPI:
+        app = FastAPI(**self._prepare_fastapi_app_params())
+        for router in self._get_fastapi_routers():
+            app.include_router(prefix=self.root_path, router=router)
+        app.dependency_overrides[depends.fastapi_endpoint_id] = partial(
+            lambda a: a, self.id
+        )
+        for exception, handler in self._get_app_exception_handlers().items():
+            app.add_exception_handler(exception, handler)
+        return app
+
+    def _get_app_exception_handlers(
+        self,
+    ) -> Dict[
+        Union[int, Type[Exception]],
+        Callable[[Request, Exception], Union[Response, Awaitable[Response]]],
+    ]:
+        """Return a dict of exception handlers to register on the app
+
+        The key is the exception class or status code to handle.
+        The value is the handler function.
+
+        If you need to register your own handler, you can do it by overriding
+        this method and calling super(). Changes done in this way will be applied
+        to all the endpoints. If you need to register a handler only for a specific
+        endpoint, you can do it by overriding the _get_app_exception_handlers method
+        and conditionally returning your specific handlers only for the endpoint
+        you want according to the self.app value.
+
+        Be careful to not forget to roll back the transaction when you implement
+        your own error handler. If you don't, the transaction will be committed
+        and the changes will be applied to the database.
+        """
+        self.ensure_one()
+        return {
+            Exception: error_handlers._odoo_exception_handler,
+            HTTPException: error_handlers._odoo_http_exception_handler,
+            odoo.exceptions.UserError: error_handlers._odoo_user_error_handler,
+            odoo.exceptions.AccessError: error_handlers._odoo_access_error_handler,
+            odoo.exceptions.MissingError: error_handlers._odoo_missing_error_handler,
+            odoo.exceptions.ValidationError: error_handlers._odoo_validation_error_handler,
+        }
+
+    def _prepare_fastapi_app_params(self) -> Dict[str, Any]:
+        """Return the params to pass to the Fast API app constructor"""
+        return {
+            "title": self.name,
+            "description": self.description,
+            "openapi_url": self.openapi_url,
+            "docs_url": self.docs_url,
+            "redoc_url": self.redoc_url,
+            "middleware": self._get_fastapi_app_middlewares(),
+            "dependencies": self._get_fastapi_app_dependencies(),
+        }
+
+    def _get_fastapi_routers(self) -> List[APIRouter]:
```

Comment:
```suggestion
    def get_fastapi_routers(self) -> List[APIRouter]:
```
IMHO, the (documentation)[https://github.com/acsone/rest-framework/blob/16.0-fastapi/fastapi/README.rst?plain=1#L222] is correct and this code must be changed

Reply:
The documentation was incorrect. This method must not be exposed over xmlrpc.